go/test/endtoend/orchestrator/orc_test.go

```diff
@@ -107,101 +126,243 @@ func createCluster(t *testing.T, numReplicas int, numRdonly int) *cluster.LocalP
 	defer cluster.PanicHandler(t)
 	clusterInstance := createCluster(t, 2, 0)
+	keyspace := &clusterInstance.Keyspaces[0]
+	shard0 := &keyspace.Shards[0]
 	defer func() {
 		clusterInstance.Teardown()
+		// Kill tablets
+		killTablets(t, shard0)
 	}()
+
+	// find master from topo
+	curMaster := shardMasterTablet(t, clusterInstance, keyspace, shard0)
+	assert.NotNil(t, curMaster, "should have elected a master")
+
+	// TODO(deepthi): we should not need to do this, the DB should be created automatically
+	_, err := curMaster.VttabletProcess.QueryTablet(fmt.Sprintf("create database IF NOT EXISTS vt_%s", keyspace.Name), keyspace.Name, false)
+	require.NoError(t, err)
+
+	var replica *cluster.Vttablet
+	for _, tablet := range shard0.Vttablets {
+		// we know we have only two tablets, so the "other" one must be the new master
+		if tablet.Alias != curMaster.Alias {
+			replica = tablet
+			break
+		}
+	}
+	// Make the replica database read-write.
+	runSQL(t, "set global read_only=OFF", replica)
+
+	// wait for repair
+	// TODO(deepthi): wait for condition instead of sleep
+	time.Sleep(15 * time.Second)
+	qr := runSQL(t, "select @@global.read_only", replica)
+	require.NotNil(t, qr)
+	require.Equal(t, 1, len(qr.Rows))
+	require.Equal(t, "[[INT64(1)]]", fmt.Sprintf("%s", qr.Rows), qr.Rows)
+}
+
+// 5. stop replication, let orc repair
+func TestStopReplication(t *testing.T) {
+	defer cluster.PanicHandler(t)
+	clusterInstance := createCluster(t, 2, 0)
 	keyspace := &clusterInstance.Keyspaces[0]
 	shard0 := &keyspace.Shards[0]
-	for _, tablet := range shard0.Vttablets {
-		// Reset status, don't wait for the tablet status. We will check it later
-		tablet.VttabletProcess.ServingStatus = ""
+	defer func() {
+		clusterInstance.Teardown()
+		// Kill tablets
+		killTablets(t, shard0)
+	}()
 
-		// Start the tablet
-		err := tablet.VttabletProcess.Setup()
-		require.NoError(t, err)
+	// find master from topo
+	curMaster := shardMasterTablet(t, clusterInstance, keyspace, shard0)
+	assert.NotNil(t, curMaster, "should have elected a master")
+
+	// TODO(deepthi): we should not need to do this, the DB should be created automatically
+	_, err := curMaster.VttabletProcess.QueryTablet(fmt.Sprintf("create database IF NOT EXISTS vt_%s", keyspace.Name), keyspace.Name, false)
+	require.NoError(t, err)
+
+	var replica *cluster.Vttablet
+	for _, tablet := range shard0.Vttablets {
+		// we know we have only two tablets, so the "other" one must be the new master
+		if tablet.Alias != curMaster.Alias {
+			replica = tablet
+			break
+		}
 	}
+	require.NotNil(t, replica, "should be able to find a replica")
+	// use vtctlclient to stop replication
+	_, err = clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("StopReplication", replica.Alias)
+	require.NoError(t, err)
+
+	// wait for repair
+	time.Sleep(15 * time.Second)
+	// check replication is setup correctly
+	checkReplication(t, clusterInstance, curMaster, []*cluster.Vttablet{replica})
```

Comment:
I feel like these tests share a lot of code in common, the code should probably be refactored to provide a framework for setup-destroy-wait-expect cycle.

Reply:
I moved as much as possible into createCluster and checkReplication. What is left is slightly different in each test case.
As we add more tests we can revisit and refactor.